game/scripts/vscripts/Sections/Chapter1/SectionCasting.ts

```diff
@@ -41,8 +49,26 @@ const start = (complete: () => void) => {
         tg.audioDialog(LocalizationKey.Script_1_BreatheFire_3, LocalizationKey.Script_1_BreatheFire_3, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
 
         tg.immediate(_ => freezePlayerHero(false)),
+        tg.immediate(ctx => listenerID = ListenToGameEvent("dota_player_used_ability", (event: DotaPlayerUsedAbilityEvent) => {
+            if (event.abilityname === abilNameBreatheFire) {
+                eventTimer = Timers.CreateTimer(2, () => {
+                    if (unitIsValidAndAlive(ctx[CustomNpcKeys.SlacksMudGolem])) {
+                        currentDialogToken = dg.playAudio(LocalizationKey.Script_1_BreatheFire_3_failed, LocalizationKey.Script_1_BreatheFire_3_failed, ctx[CustomNpcKeys.SlacksMudGolem], undefined, () => {
+                            const ability = hero.FindAbilityByName(abilNameBreatheFire)
+                            if (ability) {
+                                ability.EndCooldown()
+                            }
+                        })
+                    }
+                })
+            }
+        }, undefined)),
 
         tg.completeOnCheck(ctx => !unitIsValidAndAlive(ctx[CustomNpcKeys.SlacksMudGolem]), 0.1),
+        tg.immediate(() => {
+            currentDialogToken = undefined
```

Comment:
Could do this in the callback of playAudio above (since that's when the token becomes invalid)

Reply:
You're right, I for some reason didn't think about that ;p done!